tests/test_mbed_targets.py

```diff
@@ -67,55 +67,69 @@ def test_cannot_compare_with_non_mbed_target_instance(self):
         target = _make_mbed_target(product_code="1000")
         self.assertFalse(target == "1000")
 
-    def test_hash_is_equal_to_hash_of_product_code(self):
-        tgt = _make_mbed_target(product_code="0100")
-        self.assertEqual(tgt.__hash__(), hash(tgt.product_code))
+    def test_hash_is_equal_to_hash_of_target_properties(self):
+        tgt = _make_mbed_target(product_code="0100", platform_name="a", board_type="b")
+        self.assertEqual(tgt.__hash__(), hash(tgt.product_code) ^ hash(tgt.platform_name) ^ hash(tgt.board_type))
 
-    def test_compare_equal_hashes_equal(self):
+    def test_hash_and_eq_are_consistent(self):
         tgt_1 = _make_mbed_target(product_code="0100", board_type="a")
-        tgt_2 = _make_mbed_target(product_code="0100", board_type="b")
+        tgt_2 = _make_mbed_target(product_code="0100", board_type="a")
+        self.assertEqual(tgt_1, tgt_2)
         tgts = dict()
         tgts[tgt_1] = "test"
         self.assertEqual(tgts[tgt_2], "test")
 
+    def test_repr_string_is_correctly_formed(self):
+        tgt = _make_mbed_target(board_type="a", product_code="b", platform_name="c")
+        self.assertEqual(tgt.__repr__(), f"{tgt.__class__.__name__}(board_type=a, product_code=b, name=c)")
```

Comment:
Is `tgt.__repr__()` equivalent to `repr(tgt)`? Maybe reduce the need for dunders.
Also, I wonder if swapping `tgt.__class__.__name__` with `MbedTarget` makes sense here. 🤔